code/modules/cm_marines/equipment/kit_boxes.dm

```diff
@@ -148,52 +148,72 @@
 	new /obj/item/clothing/head/helmet/marine/specialist(src)
 	new /obj/item/clothing/suit/storage/marine/specialist(src)
 
-var/list/kits = list("Pyro" = 2, "Grenadier" = 2, "Sniper" = 2, "Scout" = 2, "Demo" = 2)
+ //-----------------SPEC KIT BOX------------------
+ //For events/WO, allows the user to choose a specalist kit out of available ones in spec_kit_boxes_left list in gloabl_lists.dm
 
-/obj/item/spec_kit //For events/WO, allowing the user to choose a specalist kit
+/obj/item/spec_kit
 	name = "specialist kit"
 	desc = "A paper box. Open it and get a specialist kit."
 	icon = 'icons/obj/items/storage.dmi'
-	icon_state = "deliverycrate"
+	icon_state = "spec_kit"
+	var/list/allowed_roles_list = list(JOB_SQUAD_SPECIALIST, JOB_WO_SQUAD_SPECIALIST)
 
-/obj/item/spec_kit/attack_self(mob/user)
-	..()
+//this one is delivered via ASRS as a reward for DEFCON/techwebs/whatever else we will have
+/obj/item/spec_kit/asrs
+	allowed_roles_list = list(JOB_SQUAD_MARINE, JOB_WO_SQUAD_MARINE)
 
-	if(!skillcheck(user, SKILL_SPEC_WEAPONS, SKILL_SPEC_ALL))
-		to_chat(user, SPAN_NOTICE("This box is not for you, give it to a specialist!"))
+/obj/item/spec_kit/examine(var/mob/user)
+	. = ..()
+	if(!ishuman(user) && !isobserver(user))
 		return
-	if(select_and_spawn(user))
-		qdel(src)
 
-/obj/item/spec_kit/asrs
-	desc = "A paper box. Open it and get a specialist kit. Works only for squad riflemen."
+	var/allowed_roles
+	if(length(allowed_roles_list))
+		allowed_roles = ""
+		for(var/role in allowed_roles_list)
+			if(length(allowed_roles))
+				allowed_roles += ", "
+			allowed_roles += SPAN_HELPFUL("[role]")
+	else
+		allowed_roles = SPAN_HELPFUL("anyone")
+	to_chat(user, SPAN_INFO("This [name] can be used by [allowed_roles] if they didn't use one of these yet."))
 
-/obj/item/spec_kit/asrs/attack_self(mob/user)
+/obj/item/spec_kit/attack_self(var/mob/living/carbon/human/user)
 	..()
+	if(!ishuman(user))
+		to_chat(user, SPAN_WARNING("You can't use \the [src]!"))
+		return
 
-	if(user.job == JOB_SQUAD_MARINE)
+	if(!length(allowed_roles_list))
 		if(select_and_spawn(user))
 			qdel(src)
-	else
-		to_chat(user, SPAN_NOTICE("This box is not for you, give it to a squad rifleman!"))
+			return
 
-/obj/item/spec_kit/proc/select_and_spawn(mob/living/carbon/human/user)
-	if(!istype(user))
-		to_chat(user, SPAN_WARNING("You can't use \the [src]!"))
-		return
+	for(var/allowed_role in allowed_roles_list)
```

Comment:
this should be made into /proc/can_use(var/mob/living/carbon/human/user) so you don't have to do select_and_spawn above and below here, something like

/proc/can_use(var/mob/living/carbon/human/user)
   if(!length(allowed_roles_list))
       return TRUE
   loop through list
      matches 
          return TRUE
   return FALSE

Reply:
Done